processor/stream/processor.go

```diff
@@ -82,43 +76,21 @@ func (s *srErrorWrapper) Read() (map[string]interface{}, error) {
 	return v, err
 }
 
+type EventModel struct {
+	Name   string
+	Schema *jsonschema.Schema
+	Decode func(interface{}, time.Time, metadata.Metadata) (transform.Transformable, error)
```

Comment:
Is there benefit in keeping these separate? Can we make the `Decode` function/method responsible for validation too? By doing that, and removing Name (see further comments), we could simplify this to an interface:

```go
type Decoder interface {
    Decode(interface{}, time.Time, metadata.Metadata) (transform.Transformable, error)
}
```

Doing this would enable the decoder to optimise validation/decoding to avoid duplicate work, which I think Silvia is investigating now.

Reply:
I guess the benefit is write it only once, and not on each `Decode` implementation.